### PR TITLE
[Logs] Revert logging API additions

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

The spec is influx so in order to unblock 1.4 release this PR reverts all of the logging API additions. I'll move this work to a dedicated branch.

## TODOs

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by alanwest** [comment]
I'm beginning to suspect you of trying to outpace @cijothomas on lines added/removed with these giant diffs 🤣.

Screen Shot 2022-09-28 at 1 45 56 PM

File: OpenTelemetry.sln [hunk 1]
```diff
@@ -222,16 +222,6 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "correlation", "docs\logs\co
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Tests.Stress.Logs", "test\OpenTelemetry.Tests.Stress.Logs\OpenTelemetry.Tests.Stress.Logs.csproj", "{4298057B-24E0-47B3-BB76-C17E81AF6B39}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Examples.LoggingExtensions", "examples\LoggingExtensions\Examples.LoggingExtensions.csproj", "{F5EFF065-7AF5-4D7D-8038-CC419ABD8777}"
-EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Extensions.Serilog", "src\OpenTelemetry.Extensions.Serilog\OpenTelemetry.Extensions.Serilog.csproj", "{0D85558E-15B9-4251-BDBD-9CB7933B57E2}"
```
Comment (alanwest) [inline OpenTelemetry.sln:227]: Why no more Serilog or EventSource extensions? Weren't they functional prototypes before, albeit with some InternalsVisibleTo hacks?

Comment (CodeBlanch) [reply OpenTelemetry.sln:227]: The way the "log appender" APIs work is they have to be fed the provider to use, or get it from DI. Originally that was `OpenTelemetryLoggerProvider` and then it became `LoggerProvider`. Basically they need some of the public API we are still sorting out.

**Comment by CodeBlanch** [comment]
@alanwest 

> I'm beginning to suspect you of trying to outpace @cijothomas on lines added/removed with these giant diffs 🤣.

You better step up your game.

File: OpenTelemetry.sln (hunk 1 above)
Comment (alanwest) [reply OpenTelemetry.sln:227]: Ok, makes sense. I'd lost track of what had been added and when.

Comment (alanwest) [reply OpenTelemetry.sln:227]: This actually makes me more aware of the impact of the event api scope creep. Makes me question if the logging SIG would be amenable to first spec'ing and stabilizing a LoggerProvider/GetLogger/Emit suite of APIs that did not contain any notion of event domain/name. Seems like a smaller and more well understood problem space to just support the notion of log appenders.

**Review by alanwest — APPROVED** [review]
LGTM. It hurts 🤕 to see all this go away... but hopefully it will be a short-lived setback.

I can also take a pass over this diff after this PR merges just for an additional validation... https://github.com/open-telemetry/opentelemetry-dotnet/compare/core-1.3.0...main

**Review by utpilla — APPROVED** [review]

File: OpenTelemetry.sln (hunk 1 above)
Comment (pdonovan) [reply OpenTelemetry.sln:227]: Is the nuget package for OpenTelemetry.Extensions.Serilog still going to be available on myget.org, or is that going to disappear until a full solution is available?

Comment (cijothomas) [reply OpenTelemetry.sln:227]: all things removed from main will live in a separate branch and we'll do daily myget publish from that. (Similar to what was done for Metrics in early days.)

Comment (CodeBlanch) [reply OpenTelemetry.sln:227]: @pdonovan Working on that branch now it should be up soon with the changes 😄

**Pull request merged**